Flutter_TDD_Architecture_Course_notes.md

```diff
@@ -1 +1,277 @@
 
+## Notes for Flutter TDD Clean Architecture Course Videos
+
+This course aims to create a Number Trivia App, and will show us how we can implement test driven development to all the applications we want to do.
+
+#### 1. Explanation & Project Structure
+> Video playlist: https://www.youtube.com/watch?v=KjE2IDphA_U&list=PLB6lc7nQ1n4iYGE_khpXRdJkJEp9WOech
+
+In the `first tutorial` an explanation is given of the importance of having all functions tested in our application.  
+It also explains the **clean architecture**.
+
+All the features of the application will be divided into three folders, **domain**, **data** and **presentation**.
+It is also explained how each of these folders works and what they will contain.
+
+<br />
+
+#### 2. Entities & Use Cases 
+> Video 2: https://www.youtube.com/watch?v=lPkWX8xFthE&list=PLB6lc7nQ1n4iYGE_khpXRdJkJEp9WOech&index=2
+
+In the `second tutorial` we start by adding the packages of the dependencies to the `pubspec.yaml` file, that will be needed to build the application.
+
+```dart
+dependencies:
+  flutter:
+    sdk: flutter
+
+  get_it: ^2.0.1
+  
+  # Bloc for state management
+  flutter_bloc: ^0.21.0
+  
+  # Value equality
+  equatable: ^0.4.0
+  
+  # Functional programming thingies
+  dartz: ^0.8.6
+  
+  # Remote API
+  connectivity: ^0.4.3+7
+  
+  http: ^0.12.0+2
+  # Local cache
+  shared_preferences: ^0.5.3+4
+
+  cupertino_icons: ^0.1.2
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  mockito: ^4.1.0
+  ```
+  
+  It is explained how the **entity** will work in this application.
+  **Entity** is related to the type of data we are going to use,
+is an object about which information must be kept. In this case the information we need to keep is the NumberTrivia.  
+ 
+The following explains what `Use Cases` will be used for.
+Use cases will store all kinds of application **logic**.
+In this case Use cases will simply collect data from a repository,`GetConcreteNumberTrivia` and `GetRandomNumberTrivia`.
+  
+  #### The `either` Type
+  
+  The **`either`** type means the function can represent two types. In this case it can be _either_ an **error** or a **number**.
+In which **Left** are the **errors** and **Right** the **numbers**.
+
+```dart
+Future<Either<Failure, NumberTrivia>>```
```

Comment:
There is a Markdown error here.
see: https://github.com/dwyl/learn-flutter/blob/2cbeebfd43c5f1cc851627b17947b175a3435a5c/Flutter_TDD_Architecture_Course_notes.md#the-either-type
![image](https://user-images.githubusercontent.com/194400/74724220-d8a93900-5233-11ea-96eb-0b5cc8faf88c.png)

Please _review_ the _rendered_ file yourself for consistency ***`before`*** assigning it for review.

Reply:
I've changed the errors.